cmd/load.go

```diff
@@ -0,0 +1,308 @@
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/metrumresearchgroup/pkgr/logger"
+	"github.com/metrumresearchgroup/pkgr/rcmd"
+	"github.com/metrumresearchgroup/pkgr/rcmd/rp"
+	"github.com/spf13/viper"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/cobra"
+	log "github.com/sirupsen/logrus"
+)
+
+// loadCmd represents the load command
+var loadCmd = &cobra.Command{
+	Use:   "load",
+	Short: "Checks that installed packages can be loaded",
+	Long: `Attempts to load user packages specified in pkgr.yml to validate that each package has been installed
+successfully and can be used. Use the --all flag to load all packages in the user-library dependency tree instead of just user-level packages.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		all := viper.GetBool("all")
+		json := viper.GetBool("json")
+
+		rs := rcmd.NewRSettings(cfg.RPath)
+
+		// Get directory to test package loads from.
+		// By default, use the same directory as the config file.
+		rDir := viper.GetString("config")
+		if rDir == "" {
+			rDir, _ = os.Getwd()
+		} else {
+			rDir = filepath.Dir(rDir)
+		}
+		rDir, err := filepath.Abs(rDir)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"error":     err,
+				"directory": rDir,
+			}).Fatal("error getting absolute Path for R directory")
+		}
+
+		load(cfg.Packages, rs, rDir, cfg.Threads, all, json)
+	},
+}
+
+func init() {
+	RootCmd.AddCommand(loadCmd)
+	loadCmd.Flags().Bool("all", false, "load user packages as well as their dependencies")
+	viper.BindPFlag("all", loadCmd.LocalFlags().Lookup("all")) //There doesn't seem to be a way to bind local flags.
+	loadCmd.Flags().Bool("json", false, "output a JSON object of package info at the end")
+	viper.BindPFlag("json", loadCmd.LocalFlags().Lookup("json")) //There doesn't seem to be a way to bind local flags.
+	// loadCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+// Attempts to load all of the packages in userPackages, and (optionally) the dependencies of those packages, in an R
+// session launched from rDir.
+func load(userPackages []string, rs rcmd.RSettings, rDir string, threads int, all, toJson bool) {
+	if toJson {
+		logger.SetLogLevel("fatal") // disable logging
+	}
+
+	log.WithFields(log.Fields{
+		"all_arg" :  all,
+		"json_arg" : toJson,
+	}).Info("attempting to load packages")
+
+	log.Info("getting relevant packages via `pkgr plan`..................")
+
+	rVersion := rcmd.GetRVersion(&rs)
+	_, installPlan, _ := planInstall(rVersion, false)
+
+	log.Info("finished getting packages from `pkgr plan`__________________")
+
+	var toLoad []string
+	if all {
+		toLoad = installPlan.GetAllPackages()
+	} else {
+		toLoad = userPackages
+	}
+
+	report := InitLoadReport(getRSessionMetadata(rs, rDir))
+
+	resultsChannel := make(chan LoadResult, len(toLoad))
+	sem := make(chan int, threads * 2)
```

Comment:
think about something like....threads *2 given ncores on system is >=4 or threads*3 - i tried this by loading a load with concurrency of only 4 and i wasn't near saturating any CPU, so there is plenty of room at the top.

Reply:
consider overriding with --load-threads=N

Reply:
Created issue #251 to address in the future